public/app/features/query/components/QueryEditorRow.test.ts

```diff
@@ -121,6 +121,20 @@ describe('filterPanelDataToQuery', () => {
       timeRange: { from: dateTime(), to: dateTime(), raw: { from: 'now-1d', to: 'now' } },
     };
 
+    const panelDataA = filterPanelDataToQuery(loadingData, 'A');
+    expect(panelDataA?.state).toBe(LoadingState.Loading);
+  });
+  it('should not set the state to error if a frame has previously returned an error, but is now still loading', () => {
```

Comment:
It's very possible I'm misreading this, but I think there's nothing about this test that ensures this is an old error vs a new error. All this is testing is regardless of whether the error happened in this panel, or another panel, if not all the queries have completed keep the state as `Loading`, right? To phrase this another way: would we still expect if Panel A fails while Panel B is still loading, should the state be set to "error" or should it be set to "loading"? This change is suggesting it should be set to "loading", which I think is a reasonable change, after all you may want to "cancel" your query and can't cancel something that is in error. 

That change will indirectly also help your bug where old errors are being kept in state (most likely in an attempt to prevent errors from "flashing" when a user refreshes a still broken query).

I wonder if it would make sense to keep both the test above and the test below this but swap the order and then rename them to something like
- should keep the panel state to Loading if it's still loading other panels and this panel has errored
- should keep the panel state to Loading if it's still loading other panels and another panel has errored

Or something like that? Words are tricky.

Reply:
You're right, thinking about it, I guess the test is really testing the current logic and not JUST the expected behavior from the user's standpoint. That is, the only way panelData A would be in this state is if it returned an error previously, but you can only know that if you follow the query result flow. This test can't ensure that we're getting the status in sequential order. 
I would say we expect both A and B to get the loading state, since both queries are run, no matter which "Run" button the user clicks. I think your suggestion to rename them makes sense then.

Reply:
Renamed them to 
`
it(should keep the state in loading until all queries are finished, even if the current query has errored)
it(should keep the state in loading until all queries are finished, if another query has errored)
`
LMK if that sounds good